dask_expr/_merge.py

```diff
@@ -247,6 +254,63 @@ def _simplify_up(self, parent):
                     return type(parent)(result)
                 return result[parent_columns]
 
+    def _combine_similar(self, root: Expr):
+        # Push projections back up to avoid performing the same merge multiple times
+
+        def _remove_projections(frame):
```

Comment:
This feels complicated enough to be a stand-alone function/utility with a comment or doc-string explaining it's behavior.

Reply:
refactored into a more general method